File: onebot-client/src/main/java/cn/evolvefield/onebot/client/listener/impl/PrivateMessageEventListener.java
package cn.evolvefield.onebot.client.listener.impl;


import cn.evolvefield.onebot.client.handler.DefaultEventHandler;
import cn.evolvefield.onebot.client.handler.Handler;
import cn.evole.onebot.sdk.event.message.PrivateMessageEvent;

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/9/14 16:29
 * Version: 1.0
 */
public class PrivateMessageEventListener extends DefaultEventHandler<PrivateMessageEvent> {
    @Override
    public void onMessage(PrivateMessageEvent privateMessage) {
        //处理逻辑
        String message = privateMessage.getMessage();
        String[] split = message.split(" ");
        String key = split[0];
        Handler<PrivateMessageEvent> handler = getHandler(key);
        if (handler != null) {
            handler.handle(privateMessage);
        }
    }
}

File: onebot-client/src/main/kotlin/cn/evolvefield/onebot/client/listener/impl/PrivateMessageEventListener.kt
package cn.evolvefield.onebot.client.listener.impl

import cn.evole.onebot.sdk.event.message.PrivateMessageEvent
import cn.evolvefield.onebot.client.handler.DefaultEventHandler

/**
 * Description:
 * Author: cnlimiter
 * Date: 2022/9/14 16:29
 * Version: 1.0
 */
class PrivateMessageEventListener : DefaultEventHandler<PrivateMessageEvent>() {
    override suspend fun onMessage(e: PrivateMessageEvent) {
        //处理逻辑
        val message = e.message
        val split = message.split(" ".toRegex()).dropLastWhile { it.isEmpty() }.toTypedArray()
        val key = split[0]
        val handler = getHandler(key)
        handler.handle(e)
    }
}